### ISSUE TITLE
Incorrect childspancount?

### ISSUE THREAD

**Issue by huikang** [open-telemetry/opentelemetry-go#396]
When I tested the `example/http/server/server.go` and `example/http/client/client.go` program, I found that the ChildSpanCount of the client seems not correct. Followings are prints from the stdout 

Client:
```
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "92f64da1ec9594cf",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.dns",
	"StartTime": "2019-12-21T22:06:54.573172-05:00",
	"EndTime": "2019-12-21T22:06:54.57418427-05:00",
	"Attributes": [
		{
			"Key": "http.host",
			"Value": {
				"Type": "STRING",
				"Value": "localhost"
			}
		}
	],
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "d969f322ad4a835d",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.connect",
	"StartTime": "2019-12-21T22:06:54.574749-05:00",
	"EndTime": "2019-12-21T22:06:54.575186809-05:00",
	"Attributes": [
		{
			"Key": "http.remote",
			"Value": {
				"Type": "STRING",
				"Value": "[::1]:7777"
			}
		}
	],
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "44edf57e75f33363",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.getconn",
	"StartTime": "2019-12-21T22:06:54.57304-05:00",
	"EndTime": "2019-12-21T22:06:54.575332134-05:00",
	"Attributes": [
		{
			"Key": "http.host",
			"Value": {
				"Type": "STRING",
				"Value": "localhost:7777"
			}
		},
		{
			"Key": "http.remote",
			"Value": {
				"Type": "STRING",
				"Value": "[::1]:7777"
			}
		},
		{
			"Key": "http.local",
			"Value": {
				"Type": "STRING",
				"Value": "[::1]:59864"
			}
		}
	],
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "c6752583f8f9cd46",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.send",
	"StartTime": "2019-12-21T22:06:54.575557-05:00",
	"EndTime": "2019-12-21T22:06:54.575560352-05:00",
	"Attributes": null,
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "4a253b78baebad51",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.receive",
	"StartTime": "2019-12-21T22:06:54.576424-05:00",
	"EndTime": "2019-12-21T22:06:54.576474832-05:00",
	"Attributes": null,
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "4cef5552616488b6",
		"TraceFlags": 1
	},
	"ParentSpanID": "0000000000000000",
	"SpanKind": 1,
	"Name": "example/client/say hello",
	"StartTime": "2019-12-21T22:06:54.572911-05:00",
	"EndTime": "2019-12-21T22:06:54.57655628-05:00",
	"Attributes": null,
	"MessageEvents": null,
	"Links": null,
	"Status": 0,
	"HasRemoteParent": false,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 7
}
Response Received: Hello, world!
```

There are 6 spans in total, and the last span is the parent span of the other 5 spans. On the server side, there is one span (as shown below) which is the child span from the client.

```
{
	"SpanContext": {
		"TraceID": "bfaf0f17c268f15e2ba6e77ea3860676",
		"SpanID": "8181ae6e6c6c15a2",
		"TraceFlags": 1
	},
	"ParentSpanID": "4cef5552616488b6",
	"Status": 0,
	"HasRemoteParent": true,
	"DroppedAttributeCount": 0,
	"DroppedMessageEventCount": 0,
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
```
So, the span ID `4cef5552616488b6` has 6 child span in title, but why the output says `"ChildSpanCount": 7`? Thanks.

**Comment by huikang** [comment]
Looks like it misses one span end call. With the PR, I got the missing one:

```
{
	"SpanContext": {
		"TraceID": "89a113290fa054599197b9770f6d4f76",
		"SpanID": "9a15bb7a5be74bcb",
		"TraceFlags": 1
	},
	"ParentSpanID": "d73d2a13c0e12da9",
	"SpanKind": 3,
	"Name": "go.opentelemetry.io/otel/plugin/httptrace/http.request",
	"StartTime": "2019-12-21T23:43:16.557559-05:00",
	"EndTime": "2019-12-21T23:43:16.557564018-05:00",
	"DroppedLinkCount": 0,
	"ChildSpanCount": 0
}
```

**Comment by jmacd** [comment]
See also https://github.com/open-telemetry/opentelemetry-specification/issues/355

The "Child count" you're seeing is the local child count, it doesn't count remotes.

**Comment by rghetia** [comment]
fixed by #397
